pkg/serve/budget_integration_test.go

```
@@ -105,40 +109,59 @@ func TestResourceBudget_DialBackoffOnAbsentSocket(t *testing.T) {
 func TestResourceBudget_DialBackoffWaitsForListener(t *testing.T) {
 	tmp := t.TempDir()
 	sockPath := filepath.Join(tmp, "delayed.sock")
 
 	var wg sync.WaitGroup
 	var listenerErr atomic.Pointer[error]
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+	wg.Go(func() {
 		time.Sleep(40 * time.Millisecond)
 		l, lerr := net.Listen("unix", sockPath)
```

golangci-lint (2.12.1)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)